runtime/rasdump/javadump.cpp

```diff
@@ -613,35 +621,123 @@ JavaCoreDumpWriter::writeTitleSection(void)
+	time_t timeNow = time(NULL);
+	struct tm utc;
+	struct tm local;
+
+	if (NULL != gmtime_r(&timeNow, &utc)) {
+		strftime(timeStamp, _MaximumTimeStampLength, TIMESTAMP_FORMAT, &utc);
+		timeStamp[_MaximumTimeStampLength] = '\0';
+
+		if (NULL != localtime_r(&timeNow, &local)) {
+			zoneAvailable = true;
+			zoneSecondsEast = (int32_t)difftime(timeNow, mktime(&utc));
+			if (0 == local.tm_isdst) {
+				zoneName = tzname[0];
+			} else if (local.tm_isdst > 0) {
+				zoneName = tzname[1];
+				/* compensate for DST because difftime() doesn't appear to do so */
+				zoneSecondsEast += 60 * 60;
+			}
+		}
+	}
+#endif  /* defined(WIN32) */
 
-	char timeStamp[_MaximumTimeStampLength];
+	_OutputStream.writeCharacters("1TIDATETIMEUTC Date: ");
+	_OutputStream.writeCharacters(timeStamp);
+	_OutputStream.writeInteger(now % 1000, ":%03d"); /* add the milliseconds */
+	_OutputStream.writeCharacters(" (UTC)\n");
+
+	{
+		RasDumpGlobalStorage *dump_storage = (RasDumpGlobalStorage *)_VirtualMachine->j9rasdumpGlobalStorage;
+		struct J9StringTokens *stringTokens = (struct J9StringTokens *)dump_storage->dumpLabelTokens;
+
+		/* lock access to the tokens */
+		omrthread_monitor_enter(dump_storage->dumpLabelTokensMutex);
+
+		j9str_set_time_tokens(stringTokens, now);
+		j9str_subst_tokens(timeStamp, _MaximumTimeStampLength, TIMESTAMP_FORMAT, stringTokens);
+		timeStamp[_MaximumTimeStampLength] = '\0';
 
-	j9str_subst_tokens(timeStamp, _MaximumTimeStampLength, "%Y/%m/%d at %H:%M:%S", stringTokens);
+		/* release access to the tokens */
+		omrthread_monitor_exit(dump_storage->dumpLabelTokensMutex);
+	}
 
 	_OutputStream.writeCharacters("1TIDATETIME    Date: ");
 	_OutputStream.writeCharacters(timeStamp);
 	_OutputStream.writeInteger(now % 1000, ":%03d"); /* add the milliseconds */
 	_OutputStream.writeCharacters("\n");
 
+	_OutputStream.writeCharacters("1TITIMEZONE    Timezone: ");
+	if (!zoneAvailable) {
+		_OutputStream.writeCharacters("(unavailable)");
+	} else {
+		_OutputStream.writeCharacters("UTC");
+		if (0 != zoneSecondsEast) {
+			const char *format = (zoneSecondsEast > 0) ? "+%d" : "-%d";
+			uint32_t offset = ((uint32_t)((zoneSecondsEast > 0) ? zoneSecondsEast : -zoneSecondsEast)) / 60;
+			uint32_t hours = offset / 60;
+			uint32_t minutes = offset % 60;
+
+			_OutputStream.writeInteger(hours, format);
+			if (0 != minutes) {
+				_OutputStream.writeInteger(minutes, ":%02d");
+			}
+		}
+		if ((NULL != zoneName) && ('\0' != *zoneName)) {
+			_OutputStream.writeCharacters(" (");
+			_OutputStream.writeCharacters(zoneName);
+			_OutputStream.writeCharacters(")");
+		}
+	}
+	_OutputStream.writeCharacters("\n");
+
 	_OutputStream.writeCharacters("1TINANOTIME    System nanotime: ");
 	_OutputStream.writeInteger64(j9time_nano_time(), "%llu");
 	_OutputStream.writeCharacters("\n");
 
 	/* Write the file name */
-	_OutputStream.writeCharacters("1TIFILENAME    Javacore filename:    ");
+	_OutputStream.writeCharacters("1TIFILENAME    Javacore filename: ");
```

Comment:
This change seems a bit risky and unnecessary, as parsing code could (but shouldn't) rely on this.

Reply:
Since the spaces don't need to be removed, and it could introduce a problem, my preference is to just leave them there.